contentgrid-spring-boot-autoconfigure/src/main/java/com/contentgrid/spring/boot/autoconfigure/data/audit/JpaAuditingAutoConfiguration.java

```diff
@@ -1,18 +1,33 @@
 package com.contentgrid.spring.boot.autoconfigure.data.audit;
 
-import com.contentgrid.spring.data.support.auditing.v1.JpaAuditingConfiguration;
+import com.contentgrid.spring.boot.autoconfigure.data.audit.JpaAuditingAutoConfiguration.JwtAuditorAwareConfiguration;
+import com.contentgrid.spring.data.support.auditing.v1.JwtAuditorAware;
+import com.contentgrid.spring.data.support.auditing.v1.UserMetadata;
 import jakarta.persistence.EntityManagerFactory;
 import org.springframework.boot.autoconfigure.AutoConfiguration;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
+import org.springframework.data.domain.AuditorAware;
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
 import org.springframework.security.oauth2.jwt.JwtClaimAccessor;
 
 @AutoConfiguration(after = {HibernateJpaAutoConfiguration.class})
 @ConditionalOnBean(EntityManagerFactory.class)
-@ConditionalOnClass({JpaAuditingConfiguration.class, JwtClaimAccessor.class})
-@Import(JpaAuditingConfiguration.class)
+@EnableJpaAuditing
+@Import(JwtAuditorAwareConfiguration.class)
 public class JpaAuditingAutoConfiguration {
 
+    @Configuration
+    @ConditionalOnClass({UserMetadata.class, JwtClaimAccessor.class})
```

Comment:
So, there will be no auditor when the `JwtClaimsAccessor` is not on the classpath.

So, when `spring-security-oauth2-jose` (dependency of `spring-boot-starter-oauth2-resource-server`) is not on the classpath, there will be no auditor; also not for anonymous users?

Reply:
There could be a default auditorAware in case when there is no `JwtClaimAccessor` on the classpath and there is still a `SecurityContextHolder` on the classpath.